src/rtp/srtp.rs

```diff
@@ -109,37 +196,65 @@ impl SrtpContext {
         header: &RtpHeader,
         srtp_index: u64, // same as ext_seq
     ) -> Option<Vec<u8>> {
-        if buf.len() < SRTP_HMAC_LEN {
-            return None;
-        }
+        match &mut self.rtp {
+            Derived::Aes128CmSha1_80 {
+                hmac, salt, dec, ..
+            } => {
+                use aes_128_cm_sha1_80::{RtpHmac, ToRtpIv, HMAC_TAG_LEN};
+
+                if buf.len() < HMAC_TAG_LEN {
+                    return None;
+                }
 
-        let hmac_start = buf.len() - SRTP_HMAC_LEN;
+                let hmac_start = buf.len() - HMAC_TAG_LEN;
 
-        if !self
-            .rtp
-            .hmac
-            .rtp_verify(&buf[..hmac_start], srtp_index, &buf[hmac_start..])
-        {
-            trace!("unprotect_rtp hmac verify fail");
-            return None;
-        }
+                if !hmac.rtp_verify(&buf[..hmac_start], srtp_index, &buf[hmac_start..]) {
+                    trace!("unprotect_rtp hmac verify fail");
```

Comment:
This should be converted to an error I think, if this happens in prod it's likely that `trace` isn't enabled which makes it hard to figure out what happened. User's can ignore the error if they want to continue executing

Reply:
Incoming broken data is not an Error. It's just sweep-under-the-rug. The internet is a dangerous place.

Reply:
I know, but errors aren't unrecoverable. The user can still ignore this error

Reply:
The Errors str0m pushes are due to incorrect user input. Corrupt data from the internet is expected input. The errors resulting from that are not user errors.